syntaxes/cshell.tmLanguage.json

```
@@ -20,15 +20,19 @@
     "keywords": {
       "patterns": [
         {
-          "match": "(?<=^|;|&|\\s)(if|elif|while|until)(?=\\()",
+          "match": "(?<=^|;|&|\\s)(if|else if|switch|while)(?= *\\(.*\\))",
```

```suggestion
          "match": "(?<=^|;|&|\\s)(if|else if|switch|while)(?=\\()",
```

Since this extension extends shellscript, the VSCode internals already handle statement blocks with whitespace between the parenthisis and the keyword:

```shell
if ( 1 -eq 1 ); then
    echo "yes"
endif
```

However, it doesn't handle something like the following (the `if(` won't be highlighted):

```shell
if( 1 -eq 1 ); then
    echo "yes"
endif
```

All this pattern is attempting to do is support the second case.

---

Ok. Agreed. See #2.
BTW: Your example code is a mixture of `sh` and `csh` ...
```csh
if ( 1 == 1 ) then
    echo "yes"
endif
```
And even github can't handle `csh` correctly. ;-)